packages/editor-ui/src/api/helpers.ts

```diff
@@ -0,0 +1,76 @@
+import axios, { AxiosRequestConfig, Method } from 'axios';
+import {
+	IDataObject,
+} from 'n8n-workflow';
+import {
+	IRestApiContext,
+} from '../Interface';
+
+
+class ResponseError extends Error {
+	// The HTTP status code of response
+	httpStatusCode?: number;
+
+	// The error code in the response
+	errorCode?: number;
+
+	// The stack trace of the server
+	serverStackTrace?: string;
+
+	/**
+	 * Creates an instance of ResponseError.
+	 * @param {string} message The error message
+	 * @param {number} [errorCode] The error code which can be used by frontend to identify the actual error
+	 * @param {number} [httpStatusCode] The HTTP status code the response should have
+	 * @param {string} [stack] The stack trace
+	 * @memberof ResponseError
+	 */
+	constructor (message: string, options?: {errorCode?: number, httpStatusCode?: number, stack?: string}) {
+		super(message);
+		this.name = 'ResponseError';
+
+		const { errorCode, httpStatusCode, stack } = options || {};
```

Comment:
Default param is [advisable](https://airbnb.io/javascript/#es6-default-parameters).
```suggestion
	constructor (message: string, options?: { errorCode?: number, httpStatusCode?: number, stack?: string } = {}) {
		super(message);
		this.name = 'ResponseError';

		const { errorCode, httpStatusCode, stack } = options;
```

Reply:
I can do this but I am not mutating params like in the airbnb examples